packages/core/react/use-chat.ts

```diff
@@ -300,12 +300,14 @@ export function useChat({
   const hookId = useId();
   const chatId = id || hookId;
 
-  // Store initial messages as a state to avoid re-rendering when using memo:
-  const [initialMessages] = useState(initialMessagesParam ?? []);
+  // Store a empty array as the initial messages
+  // (instead of using a default parameter value that gets re-created each time)
+  // to avoid re-renders:
+  const [initialMessagesFallback] = useState([]);
```

Comment:
@MaxLeiter are we expecting useChat to working in RSCs? asking bc of the `useState`

Reply:
No, it's purely a client-side hook.